app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryDescription.js

```diff
@@ -1,79 +1,74 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import Markdown from '../../Markdown';
 import { editingStoryPropTypesShape } from '../../../models/beta/story';
 import ExpandedStorySection from './ExpandedStorySection';
+import { MentionsInput, Mention } from 'react-mentions';
 
-class ExpandedStoryDescription extends React.Component {
-  constructor(props) {
-    super(props);
+const ExpandedStoryDescription = ({
+  disabled,
+  onEdit,
+  users
+}) => {
+  const [editing, setEditing] = useState(false);
 
-    this.state = {
-      editing: false
-    };
+  const toggleField = () => setEditing(!editing);
 
-    this.toggleField = this.toggleField.bind(this);
-  };
+  const formatMention = (id, display) => `@${display}`;
 
-  toggleField() {
-    this.setState({ editing: !this.state.editing });
-  };
+  const editButton = () =>
+    <button className='edit-description-button'>
+      { I18n.t('edit') }
+    </button>
 
-  editButton() {
-    return (
-      <button className='edit-description-button'>
-        { I18n.t('edit') }
-      </button>
-    );
-  };
-
-  descriptionContent(description) {
-    return (
-      <div className='markdown-wrapper'>
-        <Markdown source={description} />
-      </div>
-    );
-  };
+  const descriptionContent = description =>
+    <div className='markdown-wrapper'>
+      <Markdown source={description} />
+    </div>
 
-  descriptionTextArea(description) {
-    const { onEdit, disabled } = this.props;
+  const descriptionTextArea = description => {
+    const suggestedUsers = users.map(({ id, username }) => ({ id, display: username }));
```

Comment:
These three variables (`editButton`, `descriptionContent` and `descriptionTextArea `) are actually components, I think you could extract theses variables outside of `ExpandedStoryDescription` as components :D

Reply:
Gotcha! Thanks nolleto, I will do that.